sdk-extensions/autoconfigure-spi/src/main/java/io/opentelemetry/sdk/autoconfigure/spi/internal/StructuredConfigProperties.java

```diff
@@ -0,0 +1,169 @@
+   */
+  @Nullable
+  Long getLong(String name);
+
+  /**
+   * Returns a {@link Long} configuration property.
+   *
+   * @return a {@link Long} configuration property or {@code defaultValue} if a property with {@code
+   *     name} has not been configured
+   * @throws ConfigurationException if the property is not a valid scalar long
+   */
+  default long getLong(String name, long defaultValue) {
+    return defaultIfNull(getLong(name), defaultValue);
+  }
+
+  /**
+   * Returns a {@link Double} configuration property.
+   *
+   * @return null if the property has not been configured
+   * @throws ConfigurationException if the property is not a valid scalar double
+   */
+  @Nullable
+  Double getDouble(String name);
+
+  /**
+   * Returns a {@link Double} configuration property.
+   *
+   * @return a {@link Double} configuration property or {@code defaultValue} if a property with
+   *     {@code name} has not been configured
+   * @throws ConfigurationException if the property is not a valid scalar double
+   */
+  default double getDouble(String name, double defaultValue) {
+    return defaultIfNull(getDouble(name), defaultValue);
+  }
+
+  /**
+   * Returns a {@link List} configuration property. Empty values will be removed. Entries which are
+   * not strings are converted to their string representation.
+   *
+   * @return a {@link List} configuration property, or null if the property has not been configured
+   * @throws ConfigurationException if the property is not a valid sequence of scalars
+   */
+  @Nullable
+  List<String> getScalarList(String name);
+
+  /**
+   * Returns a {@link List} configuration property. Entries which are not strings are converted to
+   * their string representation.
+   *
+   * @see ConfigProperties#getList(String name)
+   * @return a {@link List} configuration property or {@code defaultValue} if a property with {@code
+   *     name} has not been configured
+   * @throws ConfigurationException if the property is not a valid sequence of scalars
+   */
+  default List<String> getScalarList(String name, List<String> defaultValue) {
+    return defaultIfNull(getScalarList(name), defaultValue);
+  }
+
+  /**
+   * Returns a {@link StructuredConfigProperties} configuration property.
+   *
+   * @return a map-valued configuration property, or {@code null} if {@code name} has not been
+   *     configured
+   * @throws ConfigurationException if the property is not a mapping
+   */
+  @Nullable
+  StructuredConfigProperties getStructured(String name);
+
+  /**
+   * Returns a list of {@link StructuredConfigProperties} configuration property.
+   *
+   * @return a list of map-valued configuration property, or {@code null} if {@code name} has not
+   *     been configured
+   * @throws ConfigurationException if the property is not a sequence of mappings
+   */
+  @Nullable
+  List<StructuredConfigProperties> getStructuredList(String name);
```

Comment:
`getStructured` and `getStructuredList` are the key bits which allow reading complex data.